features/dd-sdk-android-ndk/src/main/cpp/datadog-native-lib.cpp

```diff
@@ -87,7 +94,9 @@ void write_crash_report(int signum,
 void update_main_context(JNIEnv *env,
                          jstring storage_path) {
     using namespace stringutils;
-    pthread_mutex_lock(&handler_mutex);
+    if(pthread_mutex_trylock(&handler_mutex)!=0){
```

Comment:
```suggestion
    if(pthread_mutex_trylock(&handler_mutex) != 0){
```

Reply:
> [!WARNING]
> According to the [documentation](https://linux.die.net/man/3/pthread_mutex_trylock) for `pthread_mutex_trylock`, there might be several return value. Maybe it'd be relevant to actually look at the return value and act accordingly.
> 
> > If successful, the pthread_mutex_lock() and pthread_mutex_unlock() functions shall return zero; otherwise, an error number shall be returned to indicate the error.